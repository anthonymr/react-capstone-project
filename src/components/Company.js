import { NavLink } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { PropTypes } from 'prop-types';
import { fetchCompany } from '../redux/company/companySlice';

function Company({ symbol }) {
  const { company } = useSelector((state) => state.company);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCompany(symbol));
  }, [dispatch, symbol]);

  return (
    <article>
      <NavLink to="/">Home</NavLink>
      <div>{ company.symbol }</div>
      <div>{ company.companyName }</div>
      <div>{ company.price }</div>
      <div>{ company.changes }</div>
      <div>{ company.mktCap }</div>
      <div>{ company.currency }</div>
      <div>{ company.industry }</div>
      <div>{ company.description }</div>
      <div>{ company.sector }</div>
      <div>{ company.image }</div>

    </article>
  );
}

Company.propTypes = {
  symbol: PropTypes.string.isRequired,
};

export default Company;
