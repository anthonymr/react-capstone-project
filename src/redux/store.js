import { configureStore } from '@reduxjs/toolkit';
import sectorsReducer from './sectors/sectorsSlice';

const store = configureStore({
  reducer: {
    sectors: sectorsReducer,
  },
});

export default store;
