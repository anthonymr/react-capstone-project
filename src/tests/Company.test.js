import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from '../redux/store';
import Company from '../components/Company';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Provider store={store}>
          <Company symbol="BHP" />
        </Provider>,
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
