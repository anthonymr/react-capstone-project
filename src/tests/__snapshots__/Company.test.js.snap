// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <article>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </article>,
  ",",
]
`;
